### DISCUSSION TITLE
Use Heroicons from official package

### DISCUSSION THREAD

**Discussion by pedro-santiago** [laravel/nova-issues#5247] [Ideas & Feature Requests] [answered]
Today, we do have a folder called "Heroicons" with all assets inside. We could use `@heroicons/vue` instead (this would also give us tons of reshaped beautiful icons).

**Comment by davidhemphill** [comment, marked as answer]
If you're referring to v3 Heroicons, there's a ton of renamed icons, so switching over right now would likely be a breaking change.

**Reply by pedro-santiago** [reply]
Yeah, I totally agree, managed to create a component for backwards compatibility using this one: https://github.com/tailwindlabs/heroicons/issues/278#issuecomment-868966794

**Reply by pedro-santiago** [reply]
However, I'm considering an aggressive change here, since they look really really improved (v3).

**Reply by pedro-santiago** [reply]
```
<template>
  <component class="h-5 w-5" :is="getIcon(this.name)"/>
</template>


<script>
import {defineAsyncComponent} from 'vue'

export default {
  props: {
    name: {
      type: String,
      required: true,
    },
  },
  setup() {
    function getIcon(iconName) {
      return defineAsyncComponent(() => import('@heroicons/vue/24/outline/'+iconName+'Icon.js'));
    }

    return {
      getIcon
    }
  }
}
</script>```

Defined as a component called `Heroicon` and call this one from sidebar. Works like a charm ;)

**Comment by pedro-santiago** [comment]
For those who need it. It's possible, but it do takes a while since you will have to change all <Icon /> to use of that new component. Code looks neat since we can ditch all heroicons from resources folder, THUS they look reaaaaally great visually:

<img width="1138" alt="Captura de Tela 2023-01-20 às 21 28 45" src="https://user-images.githubusercontent.com/6331494/213829020-524ef1c8-225e-43e8-a1f6-1b59bfbfbe2f.png">
